include/aos/iam/certmodule.hpp

```diff
@@ -0,0 +1,386 @@
+/*
+ * Copyright (C) 2023 Renesas Electronics Corporation.
+ * Copyright (C) 2023 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef AOS_CERTMODULE_HPP_
+#define AOS_CERTMODULE_HPP_
+
+#include "aos/common/crypto.hpp"
+#include "aos/common/time.hpp"
+#include "aos/common/tools/array.hpp"
+#include "aos/common/tools/enum.hpp"
+#include "aos/common/tools/string.hpp"
+#include "aos/iam/config.hpp"
+
+namespace aos {
+namespace iam {
+namespace certhandler {
+
+/**
+ * Maximum number of module key usages.
+ */
+constexpr auto cModuleKeyUsagesMaxCount = AOS_CONFIG_CERTMODULE_KEY_USAGE_MAX_COUNT;
+
+/**
+ * Certificate type name length.
+ */
+constexpr auto cCertTypeLen = AOS_CONFIG_CERTMODULE_CERT_TYPE_NAME_LEN;
+
+/**
+ * Max number of IAM certificates per module.
+ */
+constexpr auto cCertsPerModule = AOS_CONFIG_CERTHANDLER_MAX_CERTS_PER_MODULE;
```

Comment:
It looks like this constant should be the same as `mMaxCertificates+1` in the module config. Probably we should add template parameter:

```c++
template <size_t cMaxCerts>
CertModule(...)
```